src/main/webapp/components/InputFieldText.js

```diff
@@ -8,8 +8,9 @@ export const InputFieldText = hh(class InputFieldText extends Component {
   render() {
 
     const { value, currentValue } = this.props;
-    const edited = value !== currentValue && currentValue !== undefined || this.props.valueEdited === true;
-
+    const edited = value !== currentValue && currentValue !== undefined
+      || this.props.valueEdited === true
+      || this.props.edit === true && value !== currentValue && currentValue === undefined && value !== '';
```

Comment:
Ugh, this conditional is horrible. I don't have a better suggestion than this, so I'm fine with this. It might be nice to be able to abstract these kinds of checks into a utility function. Something that checks for truthiness on a collection of objects, like `const edited = areAllTheseThingstruthy() { arguments.forEach(checkTruthiness(thing));}`

Reply:
Maybe something like this? https://lodash.com/docs/4.17.11#every

Reply:
Agree! I created some generic validations that will be util